src/AltinnCore/Common/Services/Implementation/DataSI.cs

```diff
@@ -0,0 +1,86 @@
+using System;
+using System.IO;
+using System.Net.Http;
+using System.Threading.Tasks;
+using System.Xml.Serialization;
+using AltinnCore.Common.Models;
+using AltinnCore.Common.Services.Interfaces;
+using Newtonsoft.Json;
+
+namespace AltinnCore.Common.Services.Implementation
+{
+    /// <summary>
+    /// implementation for data handling
+    /// </summary>
+    public class DataSI : IData
+    {
+        /// <summary>
+        /// Insert form for the given instance
+        /// </summary>
+        /// <typeparam name="T">The input type</typeparam>
+        /// <param name="dataToSerialize">The data to serialize</param>
+        /// <param name="instanceId">The formId</param>
+        /// <param name="type">The type</param>
+        /// <param name="applicationOwnerId">The Organization code for the service owner</param>
+        /// <param name="applicationId">The service code for the current service</param>
+        /// <param name="instanceOwnerId">The partyId</param>
+        public async Task<Instance> InsertData<T>(T dataToSerialize, Guid instanceId, Type type, string applicationOwnerId, string applicationId, int instanceOwnerId)
+        {
+            string apiUrl = $"http://localhost:5010/api/v1/instances/{instanceId}/data/boatdata?instanceOwnerId={instanceOwnerId}";
```

Comment:
Should this really be localhost?

Reply:
that is for the development environment. I didnt manage to give that a meaningful url. Jesper tried too but the routing didn't work for platform.altinn3.no.

Reply:
But how will this method work in the cluster if its using localhost?

Reply:
obs! i didnt notice that it was in code, i had thought about appsettings. Yes, this needs to be changed. I will update it